src/app/machines/views/MachineList/MachineListTable/MachineCheckbox/MachineCheckbox.tsx

```diff
@@ -10,20 +10,78 @@ import type {
   Machine,
   MachineMeta,
   MachineStateListGroup,
+  SelectedMachines,
 } from "app/store/machine/types";
 
 type Props = {
   callId?: string | null;
   label: ReactNode;
   groupValue?: MachineStateListGroup["value"];
   systemId: Machine[MachineMeta.PK];
+  machines?: Machine[];
+};
+
+const rangeSelectMachines = ({
+  systemId,
+  checked,
+  machines,
+  selected,
+}: {
+  systemId: string;
+  checked: boolean;
```

Comment:
This could be simplified and only handle check action.

```suggestion
```

uncheck action is already handled on line 127 https://github.com/canonical/maas-ui/pull/4766/files#diff-837edc7dcd4e52ceae56c014c4ef0d85cfbcdc8348f3a5c6e83ecedf9b6bdc54R127